src/Service.Tests/SqlTests/GraphQLFilterTests/MsSqlGQLFilterTests.cs

```diff
@@ -26,17 +27,22 @@ public static async Task Setup(TestContext context)
 
         /// <summary>
         /// Test Nested Filter for Many-One relationship.
+        /// Also validates authorization failure behavior when the referenced nested entity
+        /// or referenced nested entity column has permission restrictions for the given role.
         /// </summary>
-        [TestMethod]
-        public async Task TestNestedFilterManyOne()
+        [DataTestMethod]
+        [DataRow("Authenticated", false, "", DisplayName = "No nested filter AuthZ error")]
+        [DataRow("TestNestedFilterManyOne_ColumnForbidden", true, DataApiBuilderException.GRAPHQL_NESTEDFILTER_FIELD_AUTHZ_FAILURE, DisplayName = "Excluded column in nested filter entity, AuthZ failure")]
```

Comment:
Is it possible to add a test for regular filter - would it give us coverage of different code paths?

Reply:
if it follows the same path, we should be good for now